tests/Admin/Admin_Page_Tests.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Tests for the Admin_Page class.
+ *
+ * @package plugin-check
+ */
+
+namespace Admin;
+
+use WordPress\Plugin_Check\Admin\Admin_Page;
+use WP_UnitTestCase;
+
+class Admin_Page_Tests extends WP_UnitTestCase {
+
+	protected $admin_page;
+
+	public function set_up() {
+		parent::set_up();
+		$this->admin_page = new Admin_Page();
+	}
+
+	public function test_add_hooks() {
+		$this->admin_page->add_hooks();
+		$this->assertEquals( 10, has_action( 'admin_menu', array( $this->admin_page, 'add_page' ) ) );
+	}
+
+	public function test_add_page() {
+		global $_parent_pages;
+
+		$current_screen = get_current_screen();
+
+		$admin_user = self::factory()->user->create( array( 'role' => 'administrator' ) );
+
+		if ( is_multisite() ) {
+			grant_super_admin( $admin_user );
+		}
+
+		wp_set_current_user( $admin_user );
+		set_current_screen( 'dashboard' );
+
+		$this->admin_page->add_page();
+
+		$parent_pages = $_parent_pages;
+
+		set_current_screen( $current_screen );
+
+		$this->assertArrayHasKey( 'plugin-check', $parent_pages );
+		$this->assertEquals( 'tools.php', $parent_pages['plugin-check'] );
+	}
+
+	public function test_render_page() {
```

Comment:
This process of unit testing for certain strings when the template is rendered seems somewhat fragile to me, since changes to the template would also require us to update the strings these assertions are looking for. I'd prefer that we mock rendering the template file with some expected data in `$available_plugins`. We can probably override the global `$wp_object_cache` object so the 'plugins' key in the 'plugins' group contains our expected values (since that is what `get_plugins()` uses under the hood) and then confirm that the output we're getting is what we expect.

We should also include a test to ensure that we're getting the correct output whenever no plugins are availble.

Reply:
Thanks @joemcgill I've updated the tests to use the object cache to set the expected plugins and tested those. I've also included a test for when there are no plugins available.